### PR TITLE
View factory

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/angular2/src/compiler/template_compiler.ts [hunk 1]
```diff
@@ -35,6 +35,9 @@ export class TemplateCompiler {
 
   normalizeDirectiveMetadata(directive:
                                  CompileDirectiveMetadata): Promise<CompileDirectiveMetadata> {
+    if (!directive.isComponent) {
```
Comment (tbosch) [inline modules/angular2/src/compiler/template_compiler.ts]: Comment: For non components there is nothing to be normalized

**Comment by tbosch** [comment]
Reviewed in person with @IgorMinar

File: modules/angular2/src/core/compiler/template_commands.ts [hunk 2]
```diff
@@ -15,6 +15,10 @@ import {
  */
 @CONST()
 export class CompiledTemplate {
+  static getChangeDetectorFromData(data: any[]): Function { return data[0]; }
+  static getCommandsFromData(data: any[]): TemplateCmd[] { return data[1]; }
+  static getSylesFromData(data: any[]): string[] { return data[2]; }
```
Comment (gkalpak) [inline modules/angular2/src/core/compiler/template_commands.ts]: @tbosch, this was suposed to be named `getStylesFromData`, right ?

Comment (tbosch) [inline modules/angular2/src/core/compiler/template_commands.ts]: Yes, thanks for catching.

**Pull request closed without merging**